sql/schema.go

```diff
@@ -143,20 +145,13 @@ func (s *Schema) Migrate(logger *zap.Logger, db Database, before, vacuumState in
 		db.Intercept("logQueries", logQueryInterceptor(logger))
 		defer db.RemoveInterceptor("logQueries")
 	}
-	for i, m := range s.Migrations {
+	for _, m := range s.Migrations {
 		if m.Order() <= before {
 			continue
 		}
 		if err := db.WithTxImmediate(context.Background(), func(tx Transaction) error {
 			if _, ok := s.skipMigration[m.Order()]; !ok {
 				if err := m.Apply(tx, logger); err != nil {
-					for j := i; j >= 0 && s.Migrations[j].Order() > before; j-- {
-						if e := s.Migrations[j].Rollback(); e != nil {
-							err = errors.Join(err, fmt.Errorf("rollback %s: %w", m.Name(), e))
-							break
-						}
-					}
```

Comment:
I think we should keep this, because Rollback is not always a no-op.

Reply:
Reverted this part of the PR